cmd/thv/app/logs.go

```diff
@@ -2,76 +2,92 @@ package app
-				logger.Log.Errorf("failed to create container runtime: %v", err)
-				return
-			}
-
-			// List containers to find the one with the given name
-			containers, err := runtime.ListContainers(ctx)
-			if err != nil {
-				logger.Log.Errorf("failed to list containers: %v", err)
-				return
-			}
-
-			// Find the container with the given name
-			var containerID string
-			for _, c := range containers {
-				// Check if the container is managed by Vibe Tool
-				if !labels.IsToolHiveContainer(c.Labels) {
-					continue
-				}
-
-				// Check if the container name matches
-				name := labels.GetContainerName(c.Labels)
-				if name == "" {
-					name = c.Name // Fallback to container name
-				}
-
-				// Check if the name matches (exact match or prefix match)
-				if name == containerName || strings.HasPrefix(c.ID, containerName) {
-					containerID = c.ID
-					break
-				}
-			}
-
-			if containerID == "" {
-				logger.Log.Infof("container %s not found", containerName)
-				return
-			}
-
-			logs, err := runtime.ContainerLogs(ctx, containerID)
-			if err != nil {
-				logger.Log.Errorf("failed to get container logs: %v", err)
-				return
-			}
-			logger.Log.Infof(logs)
-
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return logsCmdFunc(cmd, args)
 		},
 	}
+
+	logsCommand.Flags().BoolVarP(&tailFlag, "tail", "t", false, "Tail the logs")
+	err := viper.BindPFlag("tail", logsCommand.Flags().Lookup("tail"))
+	if err != nil {
+		logger.Log.Errorf("failed to bind flag: %v", err)
+	}
+
+	return logsCommand
+}
+
+func logsCmdFunc(_ *cobra.Command, args []string) error {
+	// Get container name
+	containerName := args[0]
+
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create container runtime
+	runtime, err := container.NewFactory().Create(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create container runtime: %v", err)
+	}
+
+	// List containers to find the one with the given name
+	containers, err := runtime.ListContainers(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to list containers: %w", err)
+	}
+
+	// Find the container with the given name
+	var containerID string
+	for _, c := range containers {
+		// Check if the container is managed by Vibe Tool
+		if !labels.IsToolHiveContainer(c.Labels) {
+			continue
+		}
+
+		// Check if the container name matches
+		name := labels.GetContainerName(c.Labels)
+		if name == "" {
+			name = c.Name // Fallback to container name
+		}
+
+		// Check if the name matches (exact match or prefix match)
+		if name == containerName || strings.HasPrefix(c.ID, containerName) {
+			containerID = c.ID
+			break
+		}
+	}
+
+	if containerID == "" {
+		logger.Log.Infof("container %s not found", containerName)
+		return nil
+	}
+
+	tail := viper.GetBool("tail")
+	logs, err := runtime.ContainerLogs(ctx, containerID, tail)
+	if err != nil {
+		return fmt.Errorf("failed to get container logs: %v", err)
+	}
+	logger.Log.Infof(logs)
```

Comment:
Should this be a straight printf instead of an infof?

Reply:
yep good spot